### ISSUE TITLE
Compilation of Texture2DMS without a samplers parameter

### ISSUE THREAD

**Issue by dj2** [Microsoft/DirectXShaderCompiler#1403]
According to the Texture Object docs [1] the samplers parameter is required for Texture2DMS and Texture2DMSArray.

The below will compile correctly with DXC with optimizations turned off.
```
Texture2DMS<float4> tex2dMs;
void PSMain () {}
```

It looks like with DXIL the samples size is set to 0, which is outside the range of between 1 and 128. Should this generate a compiler error as the parameter is missing, or is the documentation incorrect?

1- https://docs.microsoft.com/en-us/windows/desktop/direct3dhlsl/dx-graphics-hlsl-to-type

**Comment by dneto0** [comment]
In SPIR-V samplers are opaque and have no values associated with them.  The sampler state is always set up on the pipeline, i.e. outside the shader.
For Vulkan, see https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VkPipelineMultisampleStateCreateInfo  which is one part of the configuration of the graphics pipeline (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#pipelines-graphics)

**Comment by dj2** [comment]
Right, so this ends up being correct SPIR-V, but is it correct HLSL? From the docs it sounds like it should just be rejected as there is no sampler count. DXIL sets the value to 0, which seems weird?

**Comment by antiagainst** [comment]
FWIW, the HLSL doc was never a definite reference for what is legal HLSL. Instead, `fxc.exe` is (at least for the current state). I've spotted multiple discrepancies between the doc and `fxc.exe` previously, for which cases we just follow `fxc.exe`.

**Comment by dneto0** [comment]
Good point.  It seems setting the value to 0 is invalid but perhaps no compiler cares.
